src/main/java/uk/gov/ons/ctp/response/sample/scheduled/distribution/SampleUnitSender.java

```diff
@@ -0,0 +1,61 @@
+package uk.gov.ons.ctp.response.sample.scheduled.distribution;
+
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.ons.ctp.common.error.CTPException;
+import uk.gov.ons.ctp.common.state.StateTransitionManager;
+import uk.gov.ons.ctp.response.sample.domain.repository.SampleUnitRepository;
+import uk.gov.ons.ctp.response.sample.message.SampleUnitPublisher;
+import uk.gov.ons.ctp.response.sample.representation.SampleUnitDTO;
+import uk.gov.ons.ctp.response.sample.representation.SampleUnitDTO.SampleUnitEvent;
+import uk.gov.ons.ctp.response.sample.representation.SampleUnitDTO.SampleUnitState;
+import uk.gov.ons.ctp.response.sampleunit.definition.SampleUnit;
+
+/** Sends SampleUnits via Rabbit queue and updates DB state */
+@Slf4j
+@Component
+public class SampleUnitSender {
+  private final SampleUnitRepository sampleUnitRepository;
+
+  private final SampleUnitPublisher sampleUnitPublisher;
+
+  private final StateTransitionManager<SampleUnitState, SampleUnitEvent>
+      sampleUnitStateTransitionManager;
+
+  public SampleUnitSender(
+      SampleUnitRepository sampleUnitRepository,
+      SampleUnitPublisher sampleUnitPublisher,
+      @Qualifier("sampleUnitTransitionManager")
+          StateTransitionManager<SampleUnitState, SampleUnitEvent>
+              sampleUnitStateTransitionManager) {
+    this.sampleUnitRepository = sampleUnitRepository;
+    this.sampleUnitPublisher = sampleUnitPublisher;
+    this.sampleUnitStateTransitionManager = sampleUnitStateTransitionManager;
+  }
+
+  /** Send a SampleUnit */
+  @Transactional(propagation = Propagation.REQUIRES_NEW)
+  public void sendSampleUnit(SampleUnit mappedSampleUnit) throws CTPException {
+    uk.gov.ons.ctp.response.sample.domain.model.SampleUnit sampleUnit =
+        sampleUnitRepository.findById(UUID.fromString(mappedSampleUnit.getId()));
+
+    // Highly unlikely, but do a quick (and inexpensive) check to make sure we haven't already
+    // delivered this Sample Unit - could happen if there's a problem with the locking/Redis
+    // timing out and releasing the lock due to the delivery process taking a long time.
+    if (sampleUnit.getState() == SampleUnitState.PERSISTED) {
+      // Send to Rabbit queue
+      sampleUnitPublisher.send(mappedSampleUnit);
```

Comment:
What API?

Reply:
Party service API. This puts a message on a queue it later consumes from and then calls an API

Reply:
nah that's a different part of the code:
PartyPublisherImpl.publish - sends Rabbit message to itself
... Rabbit internal queue ...
PartyReceiverImpl.acceptParty...

Reply:
_this_ part is sending the sample units to Collex. The part you're talking about happens a lot earlier, which could be refactored but it's not really in-scope because it happens way before Collex requests the sample units using the job API

Reply:
Oh yeah I'm being stupid today